kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaInputOperator.java

```diff
@@ -255,6 +255,10 @@ public void emitTuples()
       }
     }
     emitCount += count;
+
+    if (!consumerWrapper.doesAnyKafkaReaderThreadDied()) {
```

Comment:
This may not be the correct handling for all situations. What happens, for example, on partition or lead broker change? Or a transient connection issue? Downstream resets may be very costly and should only be triggered when really necessary.